### PR TITLE
docs(website): improve search

### LINKED ISSUE #686: Improve search on website
The search doesn't take into account the function names in the sidebar in "Reference", which makes the search experience a bit useless. For example, searching "sum_horizontal" doesn't point to `pl$sum_horizontal()` but only shows the output when we print all methods in one of the examples:

image

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #905 

It seems that the `index.json` used for search cannot take into account pieces of words, e.g `sum_hor` in `pl_sum_horizontal`, which explains why the search results were so bad. 

This PR adds custom separators when building the `index.json` for search, meaning that `pl_` and other prefixes are no longer taken into account when looking for function names. Note that `[\\s\\-]+` is the default, so only what comes after is customized.

Here are some search suggestions before and after this change.

### `sum_horizontal`

Old:

image

New:

image

### `strptime`

Old:

image

New:

image

### `sink`

Old:

image

New:

image

### CODE REVIEW

File: altdoc/mkdocs_static.yml [hunk 1]
```diff
@@ -7,7 +7,8 @@ repo_name: $ALTDOC_PACKAGE_NAME
 
 ### Plugins
 plugins:
-  - search
+  - search:
+      separator: "[\\s\\-]+|(pl_|DataFrame_|LazyFrame_|Expr_|ExprArr_|ExprBin_|ExprCat_|ExprDT_|ExprList_|ExprMeta_|ExprName_|ExprStr_|ExprStruct_)"
```
Comment (eitsupi) [inline altdoc/mkdocs_static.yml:11]: Can't it just be an underscore `_`?

Comment (etiennebacher) [reply altdoc/mkdocs_static.yml:11]: Nope, the results are not good with this when we look for functions that have a `_` like `sum_horizontal()` or `concat_str()`. I'll add a section in the preprocessing script to automatically add those prefixes.

**Comment by eitsupi** [comment]
I understand. But I don't want to close #686 because this is not a fundamental solution.
Do we need to also add `$` as a separator?

**Comment by etiennebacher** [comment]
> I don't want to close #686 because this is not a fundamental solution. 

We can close #686 later, I've updated the initial post of this PR.

> Do we need to also add `$` as a separator?

No because the search index is built on the Markdown files, which still have the `_` as separator, e.g `DataFrame_clone()`. We only replace those by `$` in the post-processing script, which is the last thing to run.

**Comment by eitsupi** [comment]
> No because the search index is built on the Markdown files, which still have the `_` as separator, e.g `DataFrame_clone()`. We only replace those by `$` in the post-processing script, which is the last thing to run.

I understand that. My point was that since $ is used in places such as examples, shouldn't that also be registered as a separator?

**Comment by etiennebacher** [comment]
> My point was that since $ is used in places such as examples, shouldn't that also be registered as a separator?

I added it, it allows a bit more matches indeed

**Review by eitsupi — APPROVED** [review]

**Pull request merged**